packages/nimble-components/src/table/styles.ts

```diff
@@ -25,6 +28,16 @@ export const styles = css`
 
     :host {
         height: 480px;
+        ${tableFitRowsHeight.cssCustomProperty}: calc(var(--ni-private-table-scroll-height) + ${controlHeight});
+        ${
+            /**
+             * Set a default maximum height for the table of 40.5 rows plus the header row so
+             * that clients don't accidentally create a table that tries to render too many rows at once.
+             * If needed, the max-height can be overridden by the client, but setting a default ensures
+             * that the max-height is considered if a larger one is needed rather than being overlooked.
+             */ ''
+        }
+        max-height: calc(${controlHeight} + (40.5 * (2 * ${borderWidth} + ${controlHeight})));
```

Comment:
Since we are unconditionally setting the `max-height` here regardless of whether or not the client is trying to use the "fit-rows" feature, do we need to consider this a breaking change or update apps to make sure full-page tables don't have their heights accidentally limited?

Thoughts? @rajsite @jattasNI

Reply:
From a [basic search](https://dev.azure.com/ni/_search?action=contents&text=%22nimble-table%20%7B%22&type=code&lp=custom-Collection&filters=&pageSize=25&result=DefaultCollection/DevCentral/srd-port/GBmaster//src/BBA/Research%20Demos/Nimble%20Demo/APM-Start/src/app/home/welcome.component.scss) looks like usages outside systemlinkshared were setting a max-height. Don't have a clear idea about what CSS changes we would consider breaking changes but I think it makes sense to lean towards caution.

So I think tasks would be:
-  Checking our usages in systemlinkshared
- Mark breaking. Do we want to mark `dependentChangeType` as breaking as well? Seems to be what we are advocating (table's have a breaking usage change so treat this upgrade more carefully, including your table usage in Angular / Blazor)
- Description should be more specific about the breaking change and mitigations. i.e., table now predefines a max-height so make sure to override if you need a specific one.

Reply:
Change has been marked as `major`, and SLE tables have been updated.